### DISCUSSION TITLE
Conditional rendering, wrap in functions

### DISCUSSION THREAD

**Discussion by askasp** [davidpdrsn/axum-live-view#152] [Q&A] [answered]
Hi! I'm coming from elixir/phoenix and really like this library. Great work!

I'm new to rust, so this question might be more rust than this library related.
However, I want to try to build a liveview app with just one large liveview impl so also routing can happen inside this impl.

However, to not  get the file to large i want to extract functions that returns Html<Self:Message>. 

The render_counter_page() func is an example of what i want to do. Ideally i want to be able to put it in another file as well.  Is there any way to achieve this? 
![image](https://user-images.githubusercontent.com/41793839/208896342-350f7c66-7666-4799-a35b-4ed7de6ea08c.png)

**Comment by davidpdrsn** [comment, marked as answer]
Yeah you can just define `fn render_counter_page(view: &YourView)`. Is that not working for you?

**Reply by askasp** [reply]
It is! Thanks :D

**Reply by askasp** [reply]
But is it somehow possible to nest a liveview in another one? Like a shared view component. It seems like the combine only works by appending one on the other

**Reply by davidpdrsn** [reply]
If they have the same message type you can do it directly otherwise you have to use combine.

**Reply by askasp** [reply]
Thanks for taking the time to answer me! As I see it combine can only make liveviews adjacent. One before the other. It does not allow you to  nest one inside the divs of another. This might not be the purpose though, i'm just trying to write the code like i would in Elm

**Reply by davidpdrsn** [reply]
Yep its not. You have to handle that yourself in your view implementation.
